ehb_datasources/drivers/redcap/driver.py

```diff
@@ -629,70 +631,90 @@ def counter(start):
                 yield start
                 start += 1
 
+        # method to identify button color and icon
+        # based on form completion status
+        def get_button_icon(key):
+            all_form_status = redcap_form_complete_codes
+            button_icon=[]
+            try:
+                if all_form_status[key] == 1: # form is unverified
+                    button_icon.append('btn-warning')
+                    button_icon.append('fa-adjust')
+                    return button_icon
+                elif all_form_status[key] == 2: # form is complete
+                    button_icon.append('btn-success')
+                    button_icon.append('fa-circle')
+                    return button_icon
+            except: # form is incomplete
+                button_icon.append('btn-primary')
+                button_icon.append('fa-circle-o')
+                return button_icon
+
         if self.form_names:
             # The project is not longitudinal
-            def makeRow(fn, i):
+            def makeRow(form_name, form_index):
+                key = str(form_index)
                 row = '<tr><td>' + reduce(
                     lambda x,
                     y: x + ' ' + y.capitalize(),
-                    fn.split('_'), '') + '</td>'
+                    form_name.split('_'), '') + '</td>'
                 return row + ('<td><button data-toggle="modal"' +
-                              ' data-backdrop="static" data-keyboard="false"' +
-                              ' href="#pleaseWaitModal" class="btn btn-small' +
```

Comment:
is it possible to update this to use the python .join function? more information can be found [here](https://www.geeksforgeeks.org/join-function-python/)

Reply:
yes - f3a23bc6e93fab826539b23658346b5cbd94ffb1